src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/msquic.cs

```diff
@@ -0,0 +1,191 @@
+#pragma warning disable IDE0073
+//
+// Copyright (c) Microsoft Corporation.
```

Comment:
@samsp-msft do we have any guidelines about including other code in our repo?
I workaround the compilation with warning disable but I got a feeling there might be some legal implications since `dotnet` belongs to the foundation and not Microsoft.

Reply:
I already mentioned I think we should find a way you don't have to check this file in, but if you do, since it's MIT licensed, that practically means you can take it and do what you want with it, which I believe, means you could change it to your normal .NET license header.

Reply:
The general guideline is at https://github.com/dotnet/runtime/blob/main/CONTRIBUTING.md#copying-files-from-other-projects

- There needs to be some details captured on where this file came from and what modifications were done to it. Either add it to the top of the file or add it in a separate file.

- This should be added to https://github.com/dotnet/runtime/blob/main/THIRD-PARTY-NOTICES.TXT

Reply:
Is the `#pragma warning disable` OK in this case or is there another way to suppress the error if I'm to leave the original notice. 

Also the example file has our notice:
https://github.com/dotnet/runtime/blob/0ae6b957ff535d86f32a315a44641361ea0e778b/src/libraries/System.Private.CoreLib/src/System/Globalization/IdnMapping.cs#L1-L2

So should I start with our notice and then the original and then info about where it came and what changes I've made?

I'll add third party notice as well, I'm mostly asking about the specifics of the imported files.

Reply:
If it is a verbatim copy of the file from the other project without any changes, it is better to leave it intact, without the .NET Foundation header. It is easier for consuming updates.

We have a lot of similar cases under https://github.com/dotnet/runtime/tree/main/src/native/external . Notice that all files in the subdirectories are verbatim copies of other projects, without the .NET Foundation header.

Reply:
it is not verbatim copy, right, @ManickaP? From the original list of changes, most made it to MsQuic, but we still change the type of exception thrown from `ThrowIfFailure`.

Reply:
It's not verbatim.

Reply:
Whats still missing? I'm happy to make it verbatim. I thought I got everything, but might have missed something.

Reply:
I see what I missed. Let me PR it into MsQuic. The only missing thing will be `using System.Net.Quic`, but that isn't actually needed from what I see.